packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1851,9 +1851,7 @@ class _TabBarViewState extends State<TabBarView> {
     super.didChangeDependencies();
     _updateTabController();
     _currentIndex = _controller!.index;
-    // TODO(chunhtai): https://github.com/flutter/flutter/issues/134253
-    _pageController?.dispose();
-    _pageController = PageController(
+    _pageController ??= PageController(
```

Comment:
you will need to update current page index. probably through a jumpTo

I think you also need to recreate PageController if viewportFraction changes, but I think it should be done in didUpdateWidget

Reply:
> you will need to update current page index. probably through a jumpTo

This is something I considered. I checked that, even without this PR, when the DefaultTabController is updated with a one with a different `initialIndex` this has a no impact on the current index.

Maybe I made a bad assumption, from my understanding `didChangeDependencies` will only be called when the `TabController` is provided via a `DefaultTabController` (because of the call to `DefaultTabController.maybeOf(context);`in `_TabBarViewState._updateTabController`. If this true, the only way to trigger `didChangeDependencies` would be similar to the one in the test I added, and there is no possibility to set a different index in this case (the current index is stored in the TabController stored in _DefaultTabControllerState).

I will do more investigation, I'm very interested to have your view on the above assumption.

> I think you also need to recreate PageController if viewportFraction changes, but I think it should be done in didUpdateWidget.

Yes, I noticed this and I planned to file another issue to give more context and a PR to fix it.

Reply:
I added a call to `jumpToPage` when the page controller is reused.